### DISCUSSION TITLE
Workaround for DCS Olympus v1.0.3

### DISCUSSION THREAD

**Discussion by TheDelta** [Aterfax/DCS-World-Dedicated-Server-Docker#75] [General]
Hey there,

I hope it's okay to use this place for feedback and instructions.
This is a guide how to get Olympus v1.0.3 working with this project.
This is possible with a docker mod & a recompiled Olympus dll's with different libraries which do work on wine.

⚠ Please note, this is experimental. Please do not report any issues to the official Olympus team yet. Use this discussion instead.

The installation should be fairly easy

## How to install

Add the variables to your `docker-compose.yml` (same procedure as with the oder docker mods, DCS Retribution and DCSServerbot)
```
      - ENABLE_DCS_OLYMPUS=${ENABLE_DCS_OLYMPUS:-0}
      - DCS_OLYMPUS_AUTOSTART_SERVER=${DCS_OLYMPUS_AUTOSTART_SERVER:-0}
      - DOCKER_MODS=thedelta/dcs-world-dedicated-server-mod-olympus:latest
````
(Note: if you use multiple docker mods, use | to add multiple mods, like `aterfax/dcs-world-dedicated-server-mod-retribution:latest|thedelta/dcs-world-dedicated-server-mod-olympus:latest`

Then update your `.env` and add
```
ENABLE_DCS_OLYMPUS=1
DCS_OLYMPUS_AUTOSTART_SERVER=1
```
(the first variable will enable Olympus installation, the 2nd will automatically start it)

Now restart your container and it should hopefully work :)
Next, you must configure Olympus to make it actually work (the autostart will fail, this is okay for now)

Simply login on web VNC (XFCE) and execute the `Olympus Configurator` shortcut. (if it's not present it's either still installing or something went wrong. Check the logs and report it)

Now configure Olympus as you wish.
(⚠ please don't use `*` as server address, it won't work! Use `0.0.0.0` instead! Next version will fix this automatically)
(📝port 3000 is probably already used, so change this!)

Execute then `Olympus Server` and it should hopefully start without an error ;)
Then execute `Olympus Client` and it should open Firefox with the Olympus web interface.

Login and enjoy Olympus (Ensure that the DCS server is running and it's not paused!)

### docker mod

- The docker mod does download Olympus v1.0.3 and the modified dll files (fix).
- nodejs will be installed
- It creates desktop shortcuts to easily interact with Olympus
  - plus workaround to open the web client in firefox
- It also fixes an issue with filenames
- (a fix for my modified dlls, because I was preparing for v1.0.4 already): link the config file to Config folder ;)

> https://github.com/TheDelta/DCS-World-Dedicated-Server-Docker (fork of this project with docker mod)

### modified dll files

- Olympus is built with cpprest, which uses windows dll calls which are not available in wine/linux
- The backend has been rewritten to use two other libraries (see fork / PR for more information)

> https://github.com/TheDelta/DCSOlympus/release (fork of Olympus)

## Whats next?

My plan is to add support out of the box without any workaround with the next version of Olympus.
And of course get the docker mod officially supported by this repository, once @Aterfax is happy with it ;)

**Comment by Aterfax** [comment]
Intending on doing some PR review for this over the weekend. Quick skim and it LGTM. Will follow up then!

**Reply by TheDelta** [reply]
Awsome. I would move the maintainer / docker tag then to your namespace, so that everything is under the same prefix.
